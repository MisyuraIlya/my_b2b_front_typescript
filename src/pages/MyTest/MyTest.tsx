import React from 'react';

const MyTest = () => {
    return (
        <div>
            <h1>Hello world</h1>
            <input placeholder='input value' type='text'/>
            <button>click me</button>
        </div>
    );
};

export default MyTest;