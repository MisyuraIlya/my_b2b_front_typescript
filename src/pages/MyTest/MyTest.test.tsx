import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import MyTest from './MyTest';

describe('TEST Component', () => {
  test('renders learn react link', () => {
    render(<MyTest />);
    const linkElement = screen.getByText(/hello world/i);
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/input value/i)
    expect(linkElement).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });
})
