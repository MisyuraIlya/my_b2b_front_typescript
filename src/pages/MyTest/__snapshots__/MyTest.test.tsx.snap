// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Component renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
